src/OrleansTestingHost/TestCluster.cs

```diff
@@ -94,6 +93,8 @@ public class TestCluster
         /// </summary>
         public SerializationManager SerializationManager { get; private set; }
 
+        public HashSet<AssemblyName> AssembliesToLoad { get; } = new HashSet<AssemblyName>();
```

Comment:
What about we utilize the new ```ApplicationParts``` infrastructure instead of this construct and populate it with parts via an extension method, instead of adding assemblies to a list what we're loading, then processing from appdomain?

Reply:
I can concat these together with `;` and set that string on the remote app domain and then explicitly load them and add them as parts but this mechanism works just as well with less code.